plugins/html-plugin/src/main/java/ru/sbtqa/tag/pagefactory/find/HtmlFindUtils.java

```diff
@@ -0,0 +1,381 @@
+        }
+        return element;
+    }
+
+    private String formErrorMessage(ComplexElement element) {
+        StringBuilder errorMessage = new StringBuilder();
+        String currenElementName = "\"" + element.getCurrentName() + "\"";
+
+        if (element.getElementPath().size() > 1) {
+            if (currenElementName.chars().allMatch(Character::isDigit)) {
+                String prevElement = element.getElementPath().get(element.getCurrentPosition() - 1).toString();
+                String elementOflistNotFound = format("with number '%s' not found in list: %s", currenElementName, prevElement);
+                errorMessage.append(elementOflistNotFound);
+            } else {
+                errorMessage.append(currenElementName);
+            }
+            errorMessage.append(". The search was performed by path: ").append(element.getFullElementPath());
+        } else {
+            errorMessage.append(currenElementName);
+        }
+        return errorMessage.toString();
+    }
+
+    /**
+     * Searches for an element of the specified type by the specified name or
+     * path
+     * <p>
+     * As a path, an enumeration of elements in a nested structure can be used
+     * through the separator "{@code ->}", for example:
+     * <ul>
+     * <li> element in the block: Block{@code ->}Element
+     * <li> element in the list with the sequence number 3: List{@code ->}3
+     * </ul>
+     * <p>
+     * If an element on the page is declared as {@code WebElement}, then an
+     * attempt will be made to convert it to the passed type if this is possible
+     * <p>
+     * By default, expects an element to appear on the page
+     *
+     * @param <T> expected element type
+     * @param name element name or path
+     * @param type element type
+     * @return Returns an element from a page by name or path
+     */
+    public <T extends WebElement> T find(String name, Class<T> type) {
+        return find(name, type, true);
+    }
+
+    /**
+     * Searches for an element of the specified type by the specified name or
+     * path
+     * <p>
+     * As a path, an enumeration of elements in a nested structure can be used
+     * through the separator "{@code ->}", for example:
+     * <ul>
+     * <li> element in the block: Block {@code ->} Element
+     * <li> element in the list with the sequence number 3: List {@code ->} 3
+     * </ul>
+     * <p>
+     * If an element on the page is declared as {@code WebElement}, then an
+     * attempt will be made to convert it to the passed type if this is possible
+     *
+     * @param <T> expected element type
+     * @param name element name or path
+     * @param type element type
+     * @param wait to wait for the element to appear or not
+     * @return Returns an element from a page by name or path
+     */
+    public <T extends WebElement> T find(String name, Class<T> type, boolean wait) {
+        Class instanсeType = type;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. ([squid:S00117](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS00117))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=133&issues=AWfPt2o64tUZZzFFj6HX&open=AWfPt2o64tUZZzFFj6HX)

[ISSUE_KEY]: AWfPt2o64tUZZzFFj6HX